### PR TITLE
Sort exports data due to memory issue

### PR DESCRIPTION
**Pull request by Offir**
[Asana Task - Mandrill data source error](https://app.asana.com/0/346935762552316/384755574729009)
There are duplicate dimensions in the original CSV that should be displayed in the end table as well.
In order to keep all the records in the DB we used dict (already_seen_map) which cause high consumption of memory.
This fix sorts exports data first so we won't need to save all the keys. Instead we 
use 2 variables (last_key, key_count).

Also pep8 fixes were added. 

TO DO: add tag version - 1.0.7  for mandrill on git + update sources.txt

### CODE REVIEW

File: panoply_mandrill/panoply_mandrill.py [hunk 1]
```diff
@@ -58,8 +63,11 @@ def __init__(self, source, opt):
         source["idpattern"] = source.get("idpattern") or IDPATTERN
 
         fromsec = int(time.time() - (DAY_RANGE * DAY))
-        # disabled since we always need 30 days back with the AllowDuplicates method
-        #self.fromTime = self.getLastTimeSucceed(source) or formatTime(time.gmtime(fromsec))
+        '''
```
Comment (liorrozen) [inline panoply_mandrill/panoply_mandrill.py]: Note the changes you made here to the comment. This was originally a line of code that was commented out but you have changed it into one big comment. 

Beyond that - if this line is always commented and is no longer needed, it should be removed. We always have the older version saved in source control so we are not loosing anything here.

File: panoply_mandrill/panoply_mandrill.py [hunk 2]
```diff
@@ -179,12 +193,13 @@ def handleOngoing(self):
         '''will handle a job that is working in batches'''
         fn = self.ongoingJob.get('function')
         return fn(self.ongoingJob)
-        
+
     def handleRequired(self, metric, required_field):
-        '''for metrics that would need an extra api call before they can work.'''
+        # for metrics that would need an extra api call before they can work.
```
Comment (liorrozen) [inline panoply_mandrill/panoply_mandrill.py]: Not a big deal, but usually, the comment underneath the function signature uses three quotes (as it was before your change). This is defined in [PEP 257](https://www.python.org/dev/peps/pep-0257/#what-is-a-docstring). 

With this comment in the correct format, python can then extract this comment and use it as the special `__doc__` attribute for the function/class/object.

File: panoply_mandrill/panoply_mandrill.py (hunk 1 above)
Comment (Offir) [reply panoply_mandrill/panoply_mandrill.py]: 👍

File: panoply_mandrill/panoply_mandrill.py (hunk 2 above)
Comment (Offir) [reply panoply_mandrill/panoply_mandrill.py]: 👍  Thanks

**Pull request merged**